code/modules/bitrunning/components/glitch.dm

```diff
@@ -0,0 +1,89 @@
+/datum/component/glitch
+	/// Ref of the spawning forge
+	var/datum/weakref/forge_ref
+
+/datum/component/glitch/Initialize(obj/machinery/quantum_server/server, obj/machinery/byteforge/forge)
+	if(!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	RegisterSignal(forge, COMSIG_MACHINERY_POWER_RESTORED, PROC_REF(on_forge_power_restored))
+	RegisterSignals(forge, list(COMSIG_MACHINERY_BROKEN, COMSIG_MACHINERY_POWER_LOST), PROC_REF(on_forge_broken))
+	forge_ref = WEAKREF(forge)
+
+	var/mob/living/owner = parent
+	server.spawned_threat_refs.Remove(WEAKREF(owner)) // so the server doesn't dust us
```

Comment:
this seems like something that should be a proc on the server imo

Reply:
this felt like an unnecessary getter/setter pattern